nsls2forge_utils/auto_tick.py

```diff
@@ -0,0 +1,627 @@
+                if not (
+                    _.startswith("recipe")
+                    or _.startswith("migrators")
+                    or _.startswith("README")
+                )
+            ]
+
+    if (
+        (
+            migrator.check_solvable
+            and feedstock_ctx.attrs["conda-forge.yml"].get("bot", {}).get("automerge")
+        )
+        or feedstock_ctx.attrs["conda-forge.yml"]
+        .get("bot", {})
+        .get("check_solvable", False)
+    ) and not is_recipe_solvable(feedstock_dir):
+        eval_cmd(f"rm -rf {feedstock_dir}")
+        return False, False
+
+    if (
+        isinstance(migrator, MigrationYaml)
+        and not diffed_files
+        and feedstock_ctx.attrs["name"] != "conda-forge-pinning"
+    ):
+        # spoof this so it looks like the package is done
+        pr_json = {
+            "state": "closed",
+            "merged_at": "never issued",
+            "id": str(uuid4()),
+        }
+    else:
+        # push up
+        try:
+            pr_json = push_repo(
+                session_ctx=migrator.ctx.session,
+                fctx=feedstock_ctx,
+                feedstock_dir=feedstock_dir,
+                body=migrator.pr_body(feedstock_ctx),
+                repo=repo,
+                title=migrator.pr_title(feedstock_ctx),
+                head=f"{migrator.ctx.github_username}:{branch_name}",
+                branch=branch_name,
+                organization=organization
+            )
+
+        # This shouldn't happen too often any more since we won't double PR
+        except github3.GitHubError as e:
+            if e.msg != "Validation Failed":
+                raise
+            else:
+                print(f"Error during push {e}")
+                # If we just push to the existing PR then do nothing to the json
+                pr_json = False
+                ljpr = False
+    if pr_json:
+        ljpr = LazyJson(
+            os.path.join(migrator.ctx.session.prjson_dir, str(pr_json["id"]) + ".json"),
+        )
+        ljpr.update(**pr_json)
+    # If we've gotten this far then the node is good
+    feedstock_ctx.attrs["bad"] = False
+    logger.info("Removing feedstock dir")
+    eval_cmd(f"rm -rf {feedstock_dir}")
+    return migrate_return, ljpr
+
+
+def initialize_migrators(github_username="", github_password="", github_token=None,
+                         dry_run=False):
+    '''
+    Setup graph, required contexts, and migrators
+
+    Parameters
+    ----------
+    github_username: str, optional
+        Username for bot on GitHub
+    github_password: str, optional
+        Password for bot on GitHub
+    github_token: str, optional
+        Token for bot on GitHub
+    dry_run: bool, optional
+        If true, does not submit pull requests on GitHub
+
+    Returns
+    -------
+    tuple
+        Migrator session to interact with GitHub,
+        temporary files, and list of migrators
+    '''
+    temp = glob.glob("/tmp/*")
```

Comment:
It would be nice to make this path configurable (via a kwarg) to potentially use some other locations.

Reply:
Agreed